packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,4 +38,156 @@ moduleFor('Syntax test: {{#with}}', class extends SharedConditionalsTest {
 
     this.assertText('False');
   }
+
+  ['@test updating the context should update the alias']() {
+    this.render(`{{#with cond1 as |cond|}}{{cond.greeting}}{{else}}False{{/with}}`, {
+      cond1: { greeting: 'Hello' }
+    });
+
+    this.assertText('Hello');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hello');
+
+    this.runTask(() => set(this.context, 'cond1', { greeting: 'Hola'}));
+
+    this.assertText('Hola');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hola');
+
+  }
+
+  ['@test the scoped variable is not available outside the {{with}} block.']() {
+    this.render(`{{name}}-{{#with other as |name|}}{{name}}{{/with}}-{{name}}`, {
+      name: 'Stef',
+      other: 'Yehuda'
+    });
+
+    this.assertText('Stef-Yehuda-Stef');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Stef-Yehuda-Stef');
```

Comment:
Yeah, I can explain it. So for this particular test, I would have written:

```javascript
['@test the scoped variable is not available outside the {{with}} block.']() {
  this.render(`{{name}}-{{#with other as |name|}}{{name}}{{/with}}-{{name}}`, {
    name: 'Stef',
    other: 'Yehuda'
  });  // (1)

  this.assertText('Stef-Yehuda-Stef');

  this.runTask(() => this.rerender()); // (2)

  this.assertText('Stef-Yehuda-Stef');

  this.runTask(() => set(this.context, other, 'Chad')); // (3)

  this.assertText('Stef-Chad-Stef');

  this.runTask(() => set(this.context, name, 'Tom')); // (4)

  this.assertText('Tom-Chad-Tom');

  this.runTask(() => {
    set(this.context, name, 'Stef');
    set(this.context, other, 'Yehuda');
  }); // (5)

  this.assertText('Stef-Yehuda-Stef');
}
```

1. Initial render
2. No-op rerender: the point is to test that "stable rerender" works (i.e. if nothing changes, we don't accidentally blow away the nodes. It could happen if we cached the wrong value, forgot to capture certain values, etc. Re-render is just a completely different path than initial render.)
3. Mutation
4. Mutation: could have groupped this with the above, too. I don't have a very strong preference for this. You could have as many of these as you need.
5. Reset: this has two purpose. Usually this is done via replacement (e.g. set(this.context, 'admin', { name: 'Tom Dale })`), but in this case, strings are immutable and there is no difference between mutation/replacement. But there is another reason – we had cases where we cached the `lastValue` in initial render and never updated it in re-render, so updating it to anything *other than* the original value would work, but you can never restore it to its original value.